data/yam/agama/auto/template.libsonnet

```
@@ -18,6 +20,9 @@ function(addon_ha_reg_code='',
   [if patterns != '' then 'software']: {
     patterns: [patterns]
   },
+  [if packages != '' then 'software']: {
```

package and patterns should be under `software` and might be one or the other, or both, so I believe this will not work.
Asking @okynos for a review.

---

Yes, need combine the condition of patterns for 'software' section.

---

```suggestion
  [if patterns != '' || packages != '' then 'software']: {
    [if patterns != '' then 'patterns']: [patterns],
    [if packages != '' then 'packages']: [packages],
```
Please review this code before committing.